Junit/src/IPL.java

```diff
@@ -0,0 +1,364 @@
+            }
+
+            reader.close();
+        }
+        catch(Exception e){
+            System.out.println("File not found "+ e);
+        }
+
+        return listOfPlayers;
+    }
+
+    // for fixing the match between team and stadium and storing in a csv file
+    public static void fixMatch(){
+        final int totalTeams = 8;
+        String[] listOfTeams = new String[totalTeams];
+        String[] homeTeamPlace = new String[totalTeams];
+
+        listOfTeams[0] =  "CSK";
+        listOfTeams[1] =  "PBKS";
+        listOfTeams[2] =  "RCB";
+        listOfTeams[3] =  "SRH";
+        listOfTeams[4] =  "MI";
+        listOfTeams[5] =  "RR";
+        listOfTeams[6] =  "DC";
+        listOfTeams[7] =  "KKR";
+
+        homeTeamPlace[0] = "HOME CSK";
+        homeTeamPlace[1] = "HOME PBKS";
+        homeTeamPlace[2] = "HOME RCB";
+        homeTeamPlace[3] = "HOME SRH";
+        homeTeamPlace[4] = "HOME MI";
+        homeTeamPlace[5] = "HOME RR";
+        homeTeamPlace[6] = "HOME DC";
+        homeTeamPlace[7] = "HOME KKR";
+
+
+        ArrayList<String> matchList = new ArrayList<String>();
+        for(int i=0;i<totalTeams;i++){
+            for(int j=i+1;j<totalTeams;j++){
+                matchList.add(listOfTeams[i] + "," + listOfTeams[j] + "," + homeTeamPlace[i] + "\n");
+                matchList.add(listOfTeams[i] + "," + listOfTeams[j] + "," + homeTeamPlace[j] + "\n");
+            }
+        }
+
+        Collections.shuffle(matchList);
+
+
+
+        try{
+            FileWriter matchWriter = new FileWriter("match.csv");
+            for (String s : matchList) {
+                matchWriter.write(s);
+            }
+            matchWriter.close();
+        }
+        catch(Exception e){
+            System.out.println(e);
+        }
+
+
+    }
+
+    // finding bowler who took 40 wickets.
+    public static void searchBowlerAtleastFortyWickets(List<Player> listOfPlayers, String team){
+
+        for(Player p: listOfPlayers){
+            if(p.getWicket() >= 40 && p.getTeam().equals(team) && p.getRole().equals("BOWLER")){
+                System.out.println(p.getName());
+            }
+        }
+    }
+
+    // searching a player by it's first name last name or full name
+    public static void searchPlayer(List<Player> listOfPlayers,String query){
+        List<Player> playerSearchList = listOfPlayers.stream().filter(p -> p.getName().equals(query) || p.getName().split(" ")[0].equals(query) || p.getName().split(" ")[1].equals(query)).collect(Collectors.toList());
+
+        for (Player player : playerSearchList) {
+            System.out.println(player.getName());
+        }
+
+    }
+
+    // find highest wicket taken or runs scored in a given team
+    public static void findHighest(List<Player> listOfPlayers,String team){
+        int higestWicketByPlayer = 0;
+        int highestRunScoredByPerson = 0;
+        int maxNameSize = 20;
+        char[] maxWicketPlayerName = new char[maxNameSize];
```

Comment:
Why store it as char array just to convert it to string while printing it? why not just store it as a string.